### PR TITLE
Made source limit a parameter

### PR DESCRIPTION
**Pull request by imbasimba**

### CODE REVIEW

File: astroquery/esasky/core.py [hunk 1]
```diff
@@ -130,6 +132,10 @@ def query_object_catalogs(self, position, catalogs=__ALL_STRING,
             Can be either a specific catalog or a list of catalogs (all catalog 
             names are found in list_catalogs()) or 'all' to search in all 
             catalogs. Defaults to 'all'. 
+        row_limit : int, optional
+            Determines how many rows that will be fetched from the database
+            for each mission. Can be -1 to select maximum (currently 100 000).
+            Defaults to 2000.
```
Comment (keflavich) [inline astroquery/esasky/core.py]: Maybe specify: "The server-set default, accessed by setting `row_limit='default'`, is 2000."  Is that correct?

File: astroquery/esasky/core.py [hunk 2]
```diff
@@ -48,6 +48,7 @@ class ESASkyClass(BaseQuery):
     __IS_SURVEY_MISSION_STRING = "isSurveyMission"
     __ZERO_ARCMIN_STRING = "0 arcmin"
     __MIN_RADIUS_CATALOG_STRING = "5 arcsec"
+    __DEFAULT_SOURCE_LIMIT = "default"
```
Comment (keflavich) [inline astroquery/esasky/core.py]: We could drop the `__` here and make this a `ConfigurationItem` and make it more accessible to users

File: astroquery/esasky/core.py [hunk 3]
```diff
@@ -465,13 +479,19 @@ def _sanitize_input_table_list(self, table_list):
...
         raise ValueError("Query_table_list must be an astropy.utils.TableList")
+    
+    def _sanitize_input_row_limit(self, row_limit):
+        if (isinstance(row_limit, int) or row_limit == self.__DEFAULT_SOURCE_LIMIT):
+            return row_limit
+        
+        raise ValueError("Row_limit must be an integer")
       
-    def _get_maps_for_mission(self, maps_table, mission, download_directory, cache):
+    def _get_maps_for_mission(self, maps_table, mission, download_dir, cache):
```
Comment (keflavich) [inline astroquery/esasky/core.py]: This was already corrected in master.  You should rebase against (or merge) master.

File: docs/esasky/esasky.rst [hunk 4]
```diff
@@ -51,7 +51,10 @@ There are two query objects methods in this module
 almost the same way except that one has catalogs as input and output and the 
 other one has mission names and observations as input and output. 
 
-For catalogs, the query returns a maximum of 2000 sources per mission.
+For catalogs, the query returns a maximum of 2000 sources per mission by 
+default. However, this can be modified by the row_limit parameter.
+You can set the parameter to -1, which will result in the maximum number of
+sources (currently 100 000).
```
Comment (keflavich) [inline docs/esasky/esasky.rst]: Would setting it to, e.g., 999999999 also work, or would the server reject that?

Comment (imbasimba) [inline docs/esasky/esasky.rst]: Higher numbers would result in the first 100 000.

**Comment by keflavich** [comment]
Looks good.  Ready to merge?

**Comment by imbasimba** [comment]
Yes :)

**Pull request merged**